File: LogicalThinkingPatterns/InvertedLetteredTrianglePattern/InvertedLetteredTriangle.py
class InvertedLetteredTriangle:
    def printPattern(self, n):
        for i in range(n, 0, -1):
            for ch in range(ord('A'), ord('A') + i):
                print(chr(ch)+" ", end ="")
            print()

    def run(self):
        n = int(input("Enter the number of rows: "))
        self.printPattern(n)

invertedLetteredTriangle = InvertedLetteredTriangle()
invertedLetteredTriangle.run()

File: LogicalThinkingPatterns/InvertedLetteredTrianglePattern/InvertedLetteredTriangle.ts
class InvertedLetteredTriangle {
  printPattern(n: number): void {
    for (let i = n; i > 0; i--) {
      let pattern: string = "";
      for (let ch = 65; ch < 65 + i; ch++) {
        pattern += String.fromCharCode(ch) + " ";
      }
      console.log(pattern);
    }
  }
}

const invertedLetteredTriangle = new InvertedLetteredTriangle();
invertedLetteredTriangle.printPattern(5);
